### PR TITLE
Add parameter to factory method for setting handler options

### PR DESCRIPTION
**Pull request by pschaeflein**
closes https://github.com/microsoft/kiota-http-dotnet/issues/233

### CODE REVIEW

File: src/KiotaClientFactory.cs [hunk 1]
```diff
@@ -32,17 +33,37 @@ public static HttpClient Create(HttpMessageHandler? finalHandler = null)
         /// Creates a default set of middleware to be used by the <see cref="HttpClient"/>.
         /// </summary>
         /// <returns>A list of the default handlers used by the client.</returns>
-        public static IList<DelegatingHandler> CreateDefaultHandlers()
+        public static IList<DelegatingHandler> CreateDefaultHandlers(IRequestOption[]? optionsForHandlers = null)
```
Comment (baywet) [inline src/KiotaClientFactory.cs:36]: Add this additional parameter to the Create method line 26 so it can be passed to the call.

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Depending on when #245 is merged, this PR will need to contain a version bump and a changelog entry.
Added a comment.
Also, did you start working on unit tests for that? Did you run into any issues with those?

**Comment by pschaeflein** [comment]
> Also, did you start working on unit tests for that? Did you run into any issues with those?

Added tests to the PR. As mentioned in #233 , I cannot run the tests locally unless I remove the analyzer reference from the .csproj file.

File: Microsoft.Kiota.Http.HttpClientLibrary.Tests/KiotaClientFactoryTests.cs [hunk 2]
```diff
@@ -87,6 +89,31 @@ public void GetDefaultHttpMessageHandlerSetsUpProxy()
...
+                if(handler is RetryHandler retryFromDefault)
+                {
+                    retryHandler = retryFromDefault;
+                    break;
+                }
+            }
+
+            // Assert
+            Assert.NotNull(retryHandler);
+            Assert.NotNull(retryHandler.RetryOption);
```
Comment (baywet) [inline Microsoft.Kiota.Http.HttpClientLibrary.Tests/KiotaClientFactoryTests.cs:114]: the option is initialized by default, here I'd rely on a reference equal with the reference you have in the arrange section instead.

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for making the changes!
since 1.4.0 has been published, we'll need an entry in the changelog (today, minor) and a version bump in the csproj to expedite the release of those changes.

**Comment by pschaeflein** [comment]
No need to expedite. Thanks tho.

File: src/KiotaClientFactory.cs [hunk 3]
```diff
@@ -47,17 +49,38 @@ public static HttpClient Create(IList<DelegatingHandler> handlers, HttpMessageHa
...
             return new List<DelegatingHandler>
             {
                 //add the default middlewares as they are ready, and add them to the list below as well
-                new UriReplacementHandler<UriReplacementHandlerOption>(),
-                new RetryHandler(),
-                new RedirectHandler(),
-                new ParametersNameDecodingHandler(),
-                new UserAgentHandler(),
-                new HeadersInspectionHandler(),
+                //add the default middlewares as they are ready
```
Comment (baywet) [inline src/KiotaClientFactory.cs:59]:
```suggestion
```

**Review by baywet — APPROVED** [review]

**Pull request merged**